README.md

```diff
@@ -1,2 +1,300 @@
+When we put the SoC to sleep, several things are turned off. This includes most of the peripherals. voltage rails, and some clocks. Basically, it draws much less power and no code is running until a wake interrupt is generated.
+
+Many Arduino projects typically have a loop where you read a sensor, do something with that reading, and then delaying for a set amount of time.
+In most cases, reading the sensor and doing something with that reading takes very little time, much smaller than the delay duration. This means that we are wasting a lot of power inside the delay doing nothing.
+By placing the SoC to sleep instead of just waiting inside the delay, we can save a considerable amount of power in most applications.
+
+
+## Tutorial #2: WakeFromIMU Example
+
+This sketch uses CuriePower library and the CurieIMU library together and demonstrates the use of an interrupt to wake the Curie SoC from sleep.
+
+```cpp
+#include <Power.h>
+#include "CurieIMU.h"
+
+
+void setup() {
+  pinMode(LED_BUILTIN, OUTPUT);
+  CurieIMU.begin();
+  CurieIMU.attachInterrupt(wakeup);
+  CurieIMU.setDetectionThreshold(CURIE_IMU_MOTION, 20);      // 100mg
+  CurieIMU.setDetectionDuration(CURIE_IMU_MOTION, 10);       // trigger times of consecutive slope data points
+  CurieIMU.interrupts(CURIE_IMU_MOTION);
+  
+}
+
+void loop() {
+  PM.sleep();
+  digitalWrite(LED_BUILTIN, HIGH);
+  delay(3000);
+  digitalWrite(LED_BUILTIN, LOW);
+}
+
+void wakeup()
+{
+  PM.wakeFromDoze();
+  // This function will be called once on device wakeup
+  // You can do some little operations here (like changing variables which will be used in the loop)
+  // Remember to avoid calling delay() and long running functions since this functions executes in interrupt context
+}
+```
+
+If you look at the beginning of loop() you will see:
+```cpp
+PM.sleep();
+```
+This line of code puts the SoC to sleep and stays in that state until it is waken up by an interrupt.
+If no interrupt triggers the SoC to wakeup, it will simply stay in a low power sleeps state forever(or until the battery runs out).
+This is specifically useful in applications that are not periodic, like in this example sketch where the SoC stays at a sleep state until the Arduino101 detects motion, or more precisely the Bosch BMI160 sensor detects the motion and triggers an interrupt to wake the Curie Soc from sleep.
+
+Inside setup() we have:
+```cpp
+CurieIMU.attachInterrupt(wakeup);
+CurieIMU.setDetectionThreshold(CURIE_IMU_MOTION, 20);      // 100mg
+CurieIMU.setDetectionDuration(CURIE_IMU_MOTION, 10);       // trigger times of consecutive slope data points
+CurieIMU.interrupts(CURIE_IMU_MOTION);
+```
+This lines of code attaches a method called wakeup() which is called whenever motion is detected. Since the interrupt signal generated by the Bosch BMI160 [6-Axis Accelerometer/Gyroscope] sensor is already internally connected to AON(Always On) interrupt of the SoC, it automatically wakes the SoC from sleep. 
```

Comment:
This is already the 2nd time you mention the BMI160. Put the extra description after the first mention, not the second.